Realm.Shared/Realm.cs

```diff
@@ -676,6 +687,56 @@ public RealmResults<dynamic> All(string className)
             return new RealmResults<dynamic>(this, metadata, true);
         }
 
+
+        #region ById
+
+        private RealmObject.Metadata PrepById<T>(out IntPtr columnIndex)
+        {
+            var metadata =  Metadata[typeof(T).Name];
+            if (metadata.ObjectIdColIndex == -1)
+                throw new RealmClassLacksObjectIdException($"Class {typeof(T).Name} does not have a property marked as ObjectId");
+            columnIndex = (IntPtr)metadata.ObjectIdColIndex;
+            return metadata;
+        }
+
+
+        /// <summary>
+        /// Fast lookup of an object from a class which has an ObjectId property.
+        /// </summary>
+        /// <typeparam name="T">The Type T must not only be a RealmObject but also have been processd by the Fody weaver, so it has persistent properties.</typeparam>
+        /// <param name="id">Id to be matched exactly, same as an == search. Int64 argument works for all integer properties supported as ObjectId.</param>
+        /// <returns>Null or an object matdhing the id.</returns>
+        /// <exception cref="RealmClassLacksObjectIdException">If the RealmObject class T lacks an [ObjectId].</exception>
+        public T ById<T>(Int64 id) where T : RealmObject
```

Comment:
I'm not convinced non-verbial method names are very .NET-friendly. Unless you've chosen it for consistency with other languages, I'd suggest going for `Find`/`GetById`.